pkg/controllermanager/controller/seed/seed_extension_check_control.go

```diff
@@ -39,8 +40,16 @@ func (c *Controller) controllerInstallationOfSeedAdd(obj interface{}) {
 	c.seedExtensionCheckQueue.Add(controllerInstallation.Spec.SeedRef.Name)
 }
 
-func (c *Controller) controllerInstallationOfSeedUpdate(_, newObj interface{}) {
-	c.controllerInstallationOfSeedAdd(newObj)
+func (c *Controller) controllerInstallationOfSeedUpdate(oldObj, newObj interface{}) {
+	oldControllerInstalaltion, ok1 := oldObj.(*gardencorev1beta1.ControllerInstallation)
+	newControllerInstalaltion, ok2 := newObj.(*gardencorev1beta1.ControllerInstallation)
+	if !ok1 || !ok2 {
+		return
+	}
+
+	if !apiequality.Semantic.DeepEqual(oldControllerInstalaltion.Status.Conditions, newControllerInstalaltion.Status.Conditions) {
```

Comment:
Should we make this check more sophisticated by only enqueuing when the `Status` of the conditions changed? Right now, even if some heart beat time or whatever time stamp will be changed we would also enqueue.

Reply:
Yep, but I would also include changes to the `Reason` and `Message` so that we get better error reporting and always reflect the actual error in the `SeedExtensions` condition